### ISSUE TITLE
react-securityGrid

### ISSUE THREAD

**Issue by mikezimm** [pnp/sp-dev-fx-webparts#1740]
## Sample (which sample are you talking about)
[React-SecurityGrid](https://github.com/pnp/sp-dev-fx-webparts/tree/master/samples/react-securitygrid)

## Authors
@russgove @hugoabernier 

## Question
I'm tagging @hugoabernier  as well because this may be a more general question.

I wanted to test the webpart out in an app catalog but I get the error saying it needs graph permissions.
Can this be done via the site collection app catalog or only on the tenant app catalog?

Everything works via gulp serve but to have have our security team review it, an installed app is better.

Thanks!  Great webpart

**Comment by hugoabernier** [comment]
@mikezimm once you upload the solution to the tenant App Catalog, they'll need to approve API permissions through the SharePoint Admin site (e.g.: yourtenant-admin.sharepoint.com).

Under **Advanced** then **API Access**, you should see requests under **Pending Requests** (it takes a minute or so to show the entries after the page is loaded). Select each request and hit **Approve** and it should work.

Once it is approved, you can continue to debug it from your `gulp serve --nobrowser`.

I hope this helps?

**Comment by PathToSharePoint** [comment]
Is this an isolated Web Part? If not, it's worth mentioning that API permissions will be granted across apps to the whole tenant. So better try it out in a dev tenant.

**Comment by mikezimm** [comment]
@PathToSharePoint , @hugoabernier ,
What do you mean by an "isolated Web Part"?

Right now when I package the solution, it should only be creating that one webpart in the solution.

**Comment by russgove** [comment]
Yeah. The solution needs no permissions. It runs on behalf of the user.

**Comment by mikezimm** [comment]
@russgove , so it just needs to API permissions through the SharePoint Admin site (e.g.: yourtenant-admin.sharepoint.com) because it's using the graph to get some information and not just SharePoint API correct?

And by doing so when it's added to the app catalog only impacts this one webpart since it's the only thing in the solution correct?

Just want to make sure I have my story correct when I try to get it added.

**Comment by russgove** [comment]
ah, sorry about that. forgot i made it look at ad groups and that uses app permissions.
it looks like i didn’t update the package.jason to reflect that
i’ll pt in a pr and send you an update tomorrow.

thanks for catching that!,
russell

**Comment by PathToSharePoint** [comment]
@mikezimm when you upload a package that requires additional permissions, you get a prompt like this:

> Components in this package require access to additional endpoints. Please contact your Tenant Administrator to ensure access to such end points is enabled. These are the permissions that need to be reviewed: Microsoft Graph, xxx.ReadWrite; Microsoft Graph, xxx.Read

These permissions will have to be approved by your tenant administrator, even if you work at the site collection level.

If a SPFx Web Part is built as "isolated", it gets its own set of approved permissions.

If not, and I guess this is the most common scenario, then the approved permissions can be used by any other app. In this case, once your tenant administrator approves the permissions, any other app can use them too. In this specific case, once Groups.Read is approved, other apps are allowed to use it too.

**Comment by russgove** [comment]
hey  christophe,
while working on this update i was trying to figure out ho to grant the required permission thru power shell, rather than a setting in  packagesolution.json.

i got it working using power shell and never updated the json file.

will push a fix tomorrow to update the json file and then users will be notifies when installing that an admin needs to grant permission.

Russell

**Comment by PathToSharePoint** [comment]
Which proves my point right? In your test, the app was able to use the permissions, even though it didn't request them directly.

**Comment by mikezimm** [comment]
@russgove @PathToSharePoint ,
I guess since I have never created anything that required Tenant admin approval...
They will want to know what they are approving.

Can you summarize in layman's terms?  I build taught myself to build webparts but I'm not a security expert or tenant admin and it's probably a good thing :)

@russgove , So the next update, I assume it will still require tenant admin approval.  And to just confirm, will that mean that any app will then be able to Read graph?  Or just this one webpart?  If not just this webpart, is there a way to make it like that or is it all or nothing?

@PathToSharePoint , yes when I ran through the workbench everything worked great.  I think that's what you were saying.

Thanks for taking the time to explain the security aspect of things.  It's very much appreciated.

**Comment by russgove** [comment]
I just did a PR for this.
Needed to add webApiPermissionRequeststo the packagesolution.json

**Comment by russgove** [comment]
The Other way to grant the permissions is to go in to azure ad and grant the permissions to the app regionstration for Sharepoint Online client Web application principal as shown below
![image](https://user-images.githubusercontent.com/2412956/110225217-3517de80-7eb1-11eb-850c-4ea84f4f9e8d.png)

**Comment by mikezimm** [comment]
@russgove , I take it your last post suggestion also applies this to any webpart, not just yours correct?

**Comment by russgove** [comment]
Yes, any share point apps you install in your tenant will be able to read all ad groups and their members
